src/olympia/addons/models.py

```diff
@@ -1307,16 +1307,30 @@ def is_recommended(self):
                 addon=self).exists()
         return recommended
 
-    @cached_property
-    def promoted_group(self):
-        """If the addon is promoted then return the Promoted group details."""
+    def is_promoted(self, group=None, application=None, currently=True):
```

Comment:
This method seems kind of complicated and convoluted, which is borne out in the test. I'm wondering why we need it to be this way. Can you describe the use cases you see for it needing to accept a `group`? I'm wondering why we cannot just have a function that returns the promoted group (as this function used to) and accepts the `application` as an argument? It seems to me like that would deliver the same data, and be less complicated.

As well, what are the use cases for `currently`?

Reply:
> Can you describe the use cases you see for it needing to accept a group? 

To make it a drop in replacement for `is_recommended = addon.is_recommended` -> `is_recommended = addon.is_promoted(group=RECOMMENDED, application=amo.FIREFOX)`.  Otherwise it's `group = addon.promoted_group(application=amo.FIREFOX); is_recommended = group and group == RECOMMENDED`.  

On the other side, if we _do_ need the group details it becomes `group = addon.promotedaddon.group if addon.is_promoted() else None`.  Do you think I should re-add `def promoted_group` that just does that?

> As well, what are the use cases for currently?

mmm, it's difficult to know because I'm writing the utility functions and model code before anything that uses it.  The API wouldn't use it.  But there are some places in the reviewer/admin tools that check `DiscoveryItem.recommendable` at the moment and `addon.is_promoted(group=RECOMMENDED, currently=False)` would be a drop in for that (possibly without the group in the future).  Though I can drop it for now and add it back later if becomes needed.  WDYT?

Reply:
I hadn't thought of all of the current uses of `is_recommended`. I think it's fine to leave as is for now, and enhance later if we find it isn't quite working for us.